src/util.h

```diff
@@ -58,9 +58,11 @@ unsigned long encodeJpeg(unsigned char **jpeg, unsigned char *buf, int width, in
 
 /* Automatically detect the file type of a given file. */
 enum filetype detectFiletype(const char *filename);
+enum filetype detectFiletypeFromBuffer(unsigned char *buf, long bufSize);
```

Comment:
Should this be part of the first patch, where the new function is introduced? Same for `decodeFileFromBuffer` below.